### ISSUE TITLE
Add Support for a 2nd Image (Drive 9)

### ISSUE THREAD

**Issue by oe3gwu** [MJoergen/C64MEGA65#93]
Your Drive 8 Support is great. But it would be nice to connect a 2nd image to drive 9, This could be a D64, or according to my other requests, a D71 or D81 or D2M Disk.

**Comment by sy2002** [comment]
@oe3gwu Thank you for your feature request. We have this on our [Roadmap](https://github.com/MJoergen/C64MEGA65/blob/master/ROADMAP.md) - and we leave this issue open to track progress.

**Comment by sy2002** [comment]
Assigned to myself for checking:

* Why has this an M2M tag? I do know that M2M theoretically supports multiple drives but I also remember there were some bugs reported during the PET development - maybe fixing the bugs in M2M is why this has an M2M tag.

* What does it take in the C64 core to support more than one drive? Back in the days I deactivated a lot of core in the MiSTer code when I implemented the 1541 - reactivating this and thinking about our general drive strategy is for sure necessary if and when we want to support more than just one 1541 - and quite some more work when we want to support `D81` and `G64` and so on

**Comment by sy2002** [comment]
The M2M part of this is a low hanging fruit. The rest is not - but also no rocket science.

### re M2M the tag of this issue:

Here are bugs in the framework (or potential bugs) related to using multiple drives that we need to fix. **Fixing these bugs are for sure low hanging fruits and are NOT changing M2M framework backwards-compatibility** so something we can and should do while V6 is being developed and then:

* https://github.com/sy2002/MiSTer2MEGA65/issues/58
* https://github.com/sy2002/MiSTer2MEGA65/issues/57
* https://github.com/sy2002/MiSTer2MEGA65/issues/52

### The actual multi-drive support (@MJoergen):

1. MiSTer "rabbithole": C64_MiSTerMEGA65/rtl/iec_drive/iec_drive.sv  as multiple simplifactions where done in past; things were commented out

2. Multiple VDrives instantiation (maybe ask Rhialto if and how this worked at PET): https://github.com/MJoergen/C64MEGA65/blob/master/CORE/vhdl/main.vhd#L1382

### Next Steps

@MJoergen I removed the research tag as the research is done: I assigned it to you: Please decide if you want to "go down the drive rabbithole" in MiSTer for V6 and if so, you can assign this one back to me first to fix the M2M part of the house.

If you want to go down the rabbit hole, then I would assume we start with a second 1541 drive ("drive 9") and are NOT talking about D81, G64, etc., yet.

**Comment by MJoergen** [comment]
@sy2002 Yes, I'm (so far) only considering having two 1541 drives. This is for sure the easiest to implement, and I think is relevant scope for V6. Anything beyond that should be postponed to later release. So, please do find the time to fix the M2M issues you've highlighted. When that is done, I'll take care of the changes to the Core.

**Comment by sy2002** [comment]
As soon as we support more than one drive, we need a setting that defines how we handle unmounted drives: Are they "disconnected/off" or "there"? Reason, as soon as you add an IEC device, it might be set to drive 9 for example. (My SD2IEC is configured like this). See here:

https://github.com/MJoergen/C64MEGA65/blob/develop/CORE/vhdl/main.vhd#L1303

We also need menu items to configure this: https://github.com/MJoergen/C64MEGA65/issues/189#issuecomment-3934003532

**Comment by sy2002** [comment]
@paich64 and note to self: We need additional test protocols for several combinations of simulated 8, 9 and real 8,9 (real drives and/or SD2IEC) and combinations of them.

**Comment by sy2002** [comment]
Let's not do this for V6. One drive is enough for now - users can augment the internal drive with hardware IEC devices. Maybe we will add D81 capabilities instead (see https://github.com/MJoergen/C64MEGA65/issues/91) and two drives would consume too much HyperRAM, so that we would not have enough HyperRAM for 2 MB `.crt` SIMCRTs on R3 machines (on R6 we could do it as we have additional SDRAM). Deferred to future versions.

**Comment by sy2002** [comment]
This will be implemented in `WIP-V6-A20` and it will support simulated 1541 and 1581 as well as using the MEGA65's floppy drive.
